gclient/src/api/calls.rs

```diff
@@ -204,6 +204,74 @@ impl GearApi {
             .await
     }
 
+    /// Migrates an active program identified by `src_program_id` onto another
+    /// node identified by `tgt_node_api` and returns the migrated program
+    /// identifier.
+    pub async fn migrate_program(
+        &self,
+        src_program_id: ProgramId,
+        tgt_node_api: &GearApi,
+    ) -> Result<ProgramId> {
+        // TODO: Ideally we have to check program is not already there (via gprog, for
+        // example). Do we want to override it? If yes, then we have to check
+        // migration is done to another node (network). How?
+        // Another solution might be that we crate a brand new ProgramId which would
+        // allow us to migrate program even withing the same node/network
```

Comment:
What are your thoughts about this? Either I have to make a check that we migrate to another node or I can generate a brand-new ProgramId and check if the code if already there then skip its migration and just clone the program

Reply:
As far as I am aware, this call is meant to be used for testing purposes by our community when they want to take a program from test/prod net and bring it into their dev net (#2055).

Reply:
In case of existance program on target destination just return error
It will require querying by program id at the begging of the function

Reply:
> I assume this `migrate_program` call is designed for users, the problem is, users are not allowed to run the sudo calls

This fn will be used for local node, where sudo calls exist

> program migration maybe require us to add a new function in pallet-gear that allows us to submit program with state

Unbelievable api, we won't add something like that cause otherwise any smart-contract platforms become meaningless